File: splitter/src/main/kotlin/com/github/telegram_bots/rss_manager/splitter/domain/Subscription.kt
package com.github.telegram_bots.rss_manager.splitter.domain

import java.sql.ResultSet

data class Subscription(val name: String, val userId: Int) {
    constructor(rs: ResultSet) : this(
            name = rs.getString("name"),
            userId = rs.getInt("user_id")
    )
}

File: core/src/main/scala/com/github/telegram_bots/telegram_rss_manager/core/domain/Subscription.scala
package com.github.telegram_bots.telegram_rss_manager.core.domain

import com.github.telegram_bots.telegram_rss_manager.core.domain.Channel.ChannelID
import com.github.telegram_bots.telegram_rss_manager.core.domain.Subscription.SubscriptionName
import com.github.telegram_bots.telegram_rss_manager.core.domain.User.UserID

case class Subscription(userId: UserID, channelId: ChannelID, name: SubscriptionName)

object Subscription {
  type SubscriptionName = String
}